org.uqbar.project.wollok.launch/src/org/uqbar/project/wollok/interpreter/listeners/WollokRemoteContextStateListener.xtend

```diff
@@ -46,17 +45,21 @@ class WollokRemoteContextStateListener implements XInterpreterListener {
 
 	override evaluated(EObject element) {
 		if (!forRepl) {
-			this.detectChanges
+			this.detectChanges(false)
 		}
 	}
 
-	def void detectChanges() {
+	def void detectChanges(boolean preserveVariables) {
 		try {
 			XDebugStackFrame.initAllVariables()
+			val Collection<XDebugStackFrameVariable> variables = if (preserveVariables) newArrayList(this.variables) else newArrayList
 			this.variables = #[]
 			val currentStack = interpreter.currentThread.stack
 			if (!currentStack.isEmpty) {
 				this.variables = new XDebugStackFrame(currentStack.peek).variables
+				if (preserveVariables) {
+					this.variables.addAll(variables)
+				}
```

Comment:
Si el if está arriba este no sería necesario, no?

Reply:
El tema es que el método que recorre las variables va pisando `this.variables` del listener, lo intenté hacer con un addAll de una y no funcionó, me quedó una lista vacía...